### PR TITLE
Cannot detect post media type: #6243

### LINKED ISSUE #6243: Cannot detect post media type. skip
Hi. 
Im working on windows and python 3.9 and latest instapy version.
my code:

from instapy import InstaPy

session = InstaPy(username="<my user>", password="<my pass>")
session.login()
session.like_by_tags(["tag"], amount=1)

### LINKED ISSUE #6277: getting Cannot detect post media type. Skip
## Expected Behavior
like a photo

## Current Behavior
doesn't like any photos 

## Possible Solution (optional)
the response from `https://www.instagram.com/explore/tags/tag/` has changed the media types are now defined with numbers not strings this might cause the issue? 

PS: Error message is: `Message: Unable to locate element: //a[@href='/p/CRT-h1RFxdq/']/child::div[@class='u7YqG']/child::span`

## InstaPy configuration
https://github.com/InstaPy/instapy-quickstart/blob/master/quickstart_templates/like_by_tag_interact_unfollow.py

### LINKED ISSUE #6350: Raspberry P4 - Cannot detect post media type
Hi, I installed on my Raspberry P4 InstaPy to test some bot solutions for instagram.
Running the program gives me this error:

Cannot detect post media type

## InstaPy configuration
Python 3.7.3
Firefox ESR

Thanks 
JK

### LINKED ISSUE #6355: Cannot detect post media type. Skip

### PR DESCRIPTION
**Pull request by elulcao**
## Description

The xpath has been updated, IG can do this all the time,
New formats has been added also, not only photos or carousel,
Fixes: #6143 #6202 #6340 #6243 #6277 #6350

```python
post_category = element.find_element_by_xpath(
                            "//a[@href='/p/"
                            + post_href.split("/")[-2]
                            + "/']/child::div[@class='u7YqG']/child::div/*[name()='svg']"
                        ).get_attribute("aria-label")
```

after fixing the xpath we should see something like:

```bash
INFO [2021-10-10 22:48:44] [internet]  Getting billie.vibes image list...
INFO [2021-10-10 22:49:12] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:12] [internet]  Found media type: Carousel or Video
INFO [2021-10-10 22:49:12] [internet]  Post category: Video
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Carousel or Video
INFO [2021-10-10 22:49:13] [internet]  Post category: Video
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Carousel or Video
INFO [2021-10-10 22:49:13] [internet]  Post category: Video
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Carousel or Video
INFO [2021-10-10 22:49:13] [internet]  Post category: Video
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Photo
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Photo
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Carousel or Video
INFO [2021-10-10 22:49:13] [internet]  Post category: Carousel
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Photo
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Photo
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Photo
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Photo
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Photo
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Photo
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Carousel or Video
INFO [2021-10-10 22:49:13] [internet]  Post category: Carousel
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Carousel or Video
INFO [2021-10-10 22:49:13] [internet]  Post category: Video
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Photo
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Carousel or Video
INFO [2021-10-10 22:49:13] [internet]  Post category: Video
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Photo
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Photo
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Photo
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Carousel or Video
INFO [2021-10-10 22:49:13] [internet]  Post category: Clip
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Photo
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Photo
INFO [2021-10-10 22:49:13] [internet]  Verifying media type: ['Photo', 'Carousel']
INFO [2021-10-10 22:49:13] [internet]  Found media type: Photo
INFO [2021-10-10 22:49:13] [internet]  Post [1/5]
```

Fixes # (issue)

## How Has This Been Tested?

Manually tested with IG acct.

- [x] Test

## Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have checked my code and corrected any misspellings
- [x] I have performed a self-review of my own code
- [x] My code follows the style guidelines of this project, `black -t py34`
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by cgonza011** [comment]
Hey im getting "Video was not able to get liked! maybe blocked?", "Invalid Like Element!", "Image was not able to get liked! maybe blocked?"

after this fix

**Comment by cgonza011** [comment]
> pip install git+https://github.com/timgrossmann/InstaPy/tree/InstaPy-6243

will try this thanks!

**Comment by elulcao** [comment]
~/InstaPy/logs/para_internet/para_internet_cookie.pkl

> im so sorry my dude...
> 
> image what does that mean?

You need to install GIT in your machine, the install InstaPy via PIP from a branch 
This is because some changes are not published to PIP, in other words pip repo is outdated and the only source of truth is GitHub 

https://git-scm.com/download/win

```bash
(venv399) ➜ ~ pip install -I git+https://github.com/timgrossmann/InstaPy.git@InstaPy-6243
Collecting git+https://github.com/timgrossmann/InstaPy.git@InstaPy-6243
  Cloning https://github.com/timgrossmann/InstaPy.git (to revision InstaPy-6243) to /private/var/folders/xy/hwplhkcj16s1cqfqb310zn700000gn/T/pip-req-build-eg7r1hj6
  Running command git clone --filter=blob:none -q https://github.com/timgrossmann/InstaPy.git /private/var/folders/xy/hwplhkcj16s1cqfqb310zn700000gn/T/pip-req-build-eg7r1hj6
  Running command git checkout -b InstaPy-6243 --track origin/InstaPy-6243
  Switched to a new branch 'InstaPy-6243'
  Branch 'InstaPy-6243' set up to track remote branch 'InstaPy-6243' from 'origin'.
  Resolved https://github.com/timgrossmann/InstaPy.git to commit f11ed89a210363c363b9517b2da179a98155b211
  Preparing metadata (setup.py) ... done
Collecting certifi>=2018.10.15
  Downloading certifi-2021.10.8-py2.py3-none-any.whl (149 kB)
     |████████████████████████████████| 149 kB 1.4 MB/s
Collecting chardet<4,>=3.0.4
```

**Review by breuerfelix — APPROVED** [review]
LGTM

**Comment by omarsameh** [comment]
> > > I did the changes in, I still get "Cannot detect post media type. Skip ..." but not on every post, is there a way to fix that??
> > 
> > 
> > Can you show me the output for logging session? I want to know the branch you are using
> 
> `INFO [2021-10-13 15:29:25] [---] Session started! OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO oooooooooooooooooooooooooooooooooooooooooooooooooooooo INFO [2021-10-13 15:29:25] [---] -- Connection Checklist [1/2] (Internet Connection Status) INFO [2021-10-13 15:29:25] [---] - Internet Connection Status: ok INFO [2021-10-13 15:29:25] [---] - Current IP is "---" and it's from "---" INFO [2021-10-13 15:29:25] [---] -- Connection Checklist [2/2] (Hide Selenium Extension) INFO [2021-10-13 15:29:25] [---] - window.navigator.webdriver response: True WARNING [2021-10-13 15:29:25] [---] - Hide Selenium Extension: error INFO [2021-10-13 15:29:34] [---] - Cookie file for user '---' loaded... ........................................................................ '''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''' INFO [2021-10-13 15:30:01] [---] Logged in successfully! INFO [2021-10-13 15:30:01] [---] Saving account progress... `

Sorry here is a better format , I removed the username, IP, and location and replaced them with "---"

INFO [2021-10-13 15:29:25] [---]  Session started!
OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO
oooooooooooooooooooooooooooooooooooooooooooooooooooooo
INFO [2021-10-13 15:29:25] [---]  -- Connection Checklist [1/2] (Internet Connection Status)
INFO [2021-10-13 15:29:25] [---]  - Internet Connection Status: ok
INFO [2021-10-13 15:29:25] [---]  - Current IP is "---" and it's from "---"
INFO [2021-10-13 15:29:25] [---]  -- Connection Checklist [2/2] (Hide Selenium Extension)
INFO [2021-10-13 15:29:25] [---]  - window.navigator.webdriver response: True
WARNING [2021-10-13 15:29:25] [---]  - Hide Selenium Extension: error
INFO [2021-10-13 15:29:34] [---]  - Cookie file for user '---' loaded...
........................................................................
''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
INFO [2021-10-13 15:30:01] [---]  Logged in successfully!
INFO [2021-10-13 15:30:01] [---]  Saving account progress...

**Comment by cgonza011** [comment]
> > > > > > Hey im getting "Video was not able to get liked! maybe blocked?", "Invalid Like Element!", "Image was not able to get liked! maybe blocked?"
> > > > > > after this fix
> > > > > 
> > > > > 
> > > > > Your account is shadow banned, or you are having a beautiful cold-down block for abusing IG limits. Wait couple of hours or delete the pkl file in the workspace, just notice that you must monitor the limits.
> > > > 
> > > > 
> > > > that is incorrect, I have not used the app what so ever and I can like my self with no issues. instapy is having an issue liking the stuff and giving out that error since the changes i made today to the files mentioned above to change.
> > > 
> > > 
> > > Could you tell me more about your configuration, which branch/commit are you using. And please show the output for the session.
> > 
> > 
> > So my install could be entirely wrong... im new at this.
> > I'm using the InstaPy-Quickstart folder. I did the whole installation from the windows "install.bat."
> > last week when I was trying to resolve the issue, I used the "pip install -I https://github.com/schealex/InstaPy/zipball/develop" to update files to the most recent ones.
> > Today I found your post about the changes to the 5 files stated above and I literally just copied and pasted the raw code from there into the corresponding files. It actually started to recognize the media types as you mentioned above and I figured everything was fixed until I noticed it was skipping every user (for having too high of following but some of the accounts had 200 followers) and if did not skip it gave the errors messages as if I had been locked out from too many likes... but this hasn't worked for me since I first installed.
> > image
> > i also want to mention that i use the "windows_start.bat" to start running the program.
> 
> Nice, nice
> 
> I was mentioning, your acct is banned by IG; a temporary cold down period. Try removing the PKL file from the workspace
> 
> `~/InstaPy/logs/user/user_cookie.pkl`
> 
> in Windows the path should be similar after deleting that file try again, Another workaround, disconnect your router (ISP) and wait ~20mins to power on. This is to generate a new IP for your router,

hey man thank you so much for your help last night unfortunately it is now back to the issue of not recognizing the image types. 

last night on my last try the cookie thing also didnt work and gave up for the night. 

any idea whats causing the issue now?

**Comment by elulcao** [comment]
like, and unlike paths updated, that was the issue for liking images
the like for comments is the same patter than like for images, and 
then more than one option was available,

update these lines, in your local to verify in the meantime the 'like' action
works as expected 

```python
xpath["like_image"] = {
    "like": "//*[contains(@class, 'fr66n')]/button/div/*[*[local-name()='svg']/@aria-label='Like']/*",
    "unlike": "//*[contains(@class, 'fr66n')]/button/div/*[*[local-name()='svg']/@aria-label='Unlike']/*",
    "play": "//*/span[contains(@aria-label, 'Play')]",
}
```

```
INFO [2021-10-16 15:04:20] [internet]  Links retrieved:: [24/https://www.instagram.com/p/CU8Rn2kq9Sf/]
INFO [2021-10-16 15:04:20] [internet]  Post [1/5]
INFO [2021-10-16 15:04:20] [internet]  https://www.instagram.com/p/CU8z1n2KwjD/
INFO [2021-10-16 15:04:23] [internet]  Image from: b'bbcmundo'
INFO [2021-10-16 15:04:23] [internet]  Image link: b'https://www.instagram.com/p/CU8z1n2KwjD/'
INFO [2021-10-16 15:04:49] [internet]  --> Image
INFO [2021-10-16 15:04:50] [internet]  --> Image liked!
INFO [2021-10-16 15:04:57] [internet]  couldn't get post description and comments
INFO [2021-10-16 15:04:57] [internet]  Post [2/5]
INFO [2021-10-16 15:04:57] [internet]  https://www.instagram.com/p/CVDCJi8KPuP/
INFO [2021-10-16 15:05:00] [internet]  Image from: b'bbcmundo'
INFO [2021-10-16 15:05:00] [internet]  Image link: b'https://www.instagram.com/p/CVDCJi8KPuP/'
INFO [2021-10-16 15:05:00] [internet]  --> Found 'Play' button for a video, trying to like it
INFO [2021-10-16 15:05:02] [internet]  --> Video
INFO [2021-10-16 15:05:04] [internet]  --> Video liked!
INFO [2021-10-16 15:05:12] [internet]  couldn't get post description and comments
INFO [2021-10-16 15:05:12] [internet]  Post [3/5]
INFO [2021-10-16 15:05:12] [internet]  https://www.instagram.com/p/CVAbGLkK16r/
INFO [2021-10-16 15:05:15] [internet]  Image from: b'bbcmundo'
INFO [2021-10-16 15:05:15] [internet]  Image link: b'https://www.instagram.com/p/CVAbGLkK16r/'
INFO [2021-10-16 15:05:42] [internet]  --> Image
INFO [2021-10-16 15:05:43] [internet]  --> Image liked!
INFO [2021-10-16 15:05:49] [internet]  couldn't get post description and comments
INFO [2021-10-16 15:05:49] [internet]  Post [4/5]
INFO [2021-10-16 15:05:49] [internet]  https://www.instagram.com/p/CVDJmSjK3oS/
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,13 @@ The **goal** of this file is explaining to the users of our project the notable
 
 _The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html)_
 
+## [0.6.25] - UNRELEASED
+
+### Fixed
+
```
Comment (breuerfelix) [inline CHANGELOG.md:10]: can you add these to 0.6.14 ? so we can make a new version once the pipelines are working :)

Comment (elulcao) [reply CHANGELOG.md:10]: Old commits were lost, I could find them in my local 
btw, Im bumping the version to .14 to continue the version path.

Comment (breuerfelix) [reply CHANGELOG.md:10]: no they are not lost, i just combined all of them together into version 0.14 :) i will merge this now and prepare a 0.14 tag!

**Pull request merged**